scanrepository/scanrepository.go

```diff
@@ -599,7 +599,11 @@ func (cfp *ScanRepositoryCmd) addVulnerabilityToFixVersionsMap(vulnerability *fo
 	} else {
 		isDirectDependency, err := utils.IsDirectDependency(vulnerability.ImpactPaths)
 		if err != nil {
-			return err
+			if cfp.scanDetails.AllowPartialResults() {
+				log.Warn(fmt.Sprintf("An empty impact path was provided for '%s'. Since partial-results is allowed the vulnerability is considered as indirect dependency and will not be fixed", vulnerability.ImpactedDependencyName))
```

Comment:
```suggestion
			if cfp.scanDetails.AllowPartialResults() {
				log.Warn(fmt.Sprintf("An empty impact path was provided for '%s'. Since partial-results is allowed the vulnerability will not be fixed", vulnerability.ImpactedDependencyName))
				log.Debug(fmt.Sprintf("Error: %s", err.Error())
```
You ignore the err without printing it.